### ISSUE TITLE
[FeatureRequest] Convert html src img urls to embeddet base64 string within the htm file

### ISSUE THREAD

**Issue by apg1980** [Fmstrat/ownnote#7]
If you copy for example the content of a website to the ownnote via copy and paste, the content will be copyed properly with the image tags of the websit which have as image src the full url to the images of the website.

it would be great to grep the image src and convert it to base64 (like your embedded editor images) to be a local source of the htm file in the ownnote.

Also all already created notes with external img sources should be converted to local base64 image src.

i will explain my post. i exported some notes of my evernotes account. i can copy and paste the notes successfully. but after i log out of my evernote account the images did not appear anymore in case of not authenticated image url.

**Comment by Fmstrat** [comment]
Can you export Evernote to HTML, rename and use them that way? I still have this on my enhancements list as I think it's a good idea regardless of Evernote.

**Comment by Fmstrat** [comment]
Merged issue: https://github.com/Fmstrat/ownnote/issues/5

**Comment by Fmstrat** [comment]
As a note: To handle this and pastes/imports from Evernote, I will implement a system that uses Javascript to grab anything in the "src" attributes. This way, if you're logged into Evernote, your browser will be able to grab the images. If I tried to do this on the server side with PHP, it would not be able to access the images.

**Comment by Fmstrat** [comment]
Update: Javascript implementation is complete (not yet on GitHub) but I've run into an issue. Most sites don't allow cross-site scripting. This means I can't load an image from many hosting sites for security reasons. I also won't be able to load Evernote images from PHP because the ownCloud server doesn't have access to your Evernote login.

Unfortunately I don't see a great way to do this. I MIGHT be able to get it to read local files, meaning you could export your Evernote files to HTML, and then copy those and your image folders into the Notes directory, then as you load each note it would convert the images to Base64. A bit clunky, though.

**Comment by apg1980** [comment]
to upload the exported htmls n folders should be the best way and convert it owncloud side. after convert delete image folders.

**Comment by maxime-roy** [comment]
Hi ! 
I don't know if it could be a solution, but if you make an export of a page in MHTML and then make it readable on your app , you won't have to bother with the servers that doesn't want cross-site scripting.

**Comment by Fmstrat** [comment]
Evernote (and any html with image folders) import is completed and will be included in the next release going out shortly. If you're following this, please be sure to + the score on apps.owncloud.com. Thanks!

**Comment by muramasatheninja** [comment]
One option I found is to use the 'SingleFile' plugin for Chrome.
https://chrome.google.com/webstore/detail/singlefile/mpiodijhokgodhhofbcjdecpffjipkle?hl=en
This exstension takes a whole page and converts it to a file with the images imbedded (Base64 Data URI if i'm using the right term) I found that I can edit the resulting page in ownnote. Also I can copy and paste straight to ownnote and it will grab images and all. He also has a github account which can be found here.
https://github.com/gildas-lormeau/SingleFile
This might be helpful to people like me who are searching for a way to quickly archive notes.
